gui/browsable/src/RFieldProvider.hxx

```diff
@@ -12,43 +12,40 @@
 #include "TH1.h"
 #include "TMath.h"
 #include <map>
+#include <memory>
 #include <string>
+#include <utility>
 
 #include <ROOT/Browsable/RProvider.hxx>
 
-#include <ROOT/RNTuple.hxx>
-// #include <ROOT/RNTupleDescriptor.hxx>
-// #include <ROOT/RMiniFile.hxx>
+#include <ROOT/RFieldVisitor.hxx>
+#include <ROOT/RPageStorage.hxx>
+#include <ROOT/RNTupleView.hxx>
 
 #include "RFieldHolder.hxx"
 
 using namespace ROOT::Experimental::Browsable;
 
 using namespace std::string_literals;
 
+template<typename T>
+using RField = ROOT::Experimental::RField<T>;
 
-// ==============================================================================================
+namespace {
 
-/** \class RFieldProvider
-\ingroup rbrowser
-\brief Base class for provider of RNTuple drawing
-\author Sergey Linev <S.Linev@gsi.de>
-\date 2021-03-09
-\warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback is welcome!
-*/
-
-class RFieldProvider : public RProvider {
-public:
-
-   // virtual ~RFieldProvider() = default;
+class RDrawVisitor : public ROOT::Experimental::Detail::RFieldVisitor {
+private:
+   std::shared_ptr<ROOT::Experimental::Detail::RPageSource> fNtplSource;
+   std::unique_ptr<TH1> fHist;
 
    /** Test collected entries if it looks like integer values and one can use better binning */
-   TH1F *TestHistBuffer(TH1F *hist)
+   void TestHistBuffer()
```

Comment:
Shouldn't this code be in the 'test' directory?

Reply:
This is not a unit test but it checks for certain special cases and adjust the histogram accordingly.